### PR TITLE
feat: allow child or loader render while redirecting to app

### PR DESCRIPTION
**Pull request by abusada**
Fixes https://github.com/gladly-team/next-firebase-auth/issues/364

### CODE REVIEW

File: src/__tests__/withAuthUser.test.js [hunk 1]
```diff
@@ -826,6 +826,29 @@ describe('withAuthUser: rendering/redirecting', () => {
     expect(container.firstChild).toBeNull()
```
Comment (kmjennison) [inline src/__tests__/withAuthUser.test.js:826]: In the above test, change description to "renders null by default when redirecting to the app".

File: src/__tests__/withAuthUser.test.js [hunk 2]
```diff
@@ -826,6 +826,29 @@ describe('withAuthUser: rendering/redirecting', () => {
...
+    const { queryByText } = render(
+      <MockCompWithUser
+        serializedAuthUser={MockSerializedAuthUser}
+        message="How are you?"
+      />
+    )
+    expect(queryByText('Hello! How are you?')).toBeTruthy()
+  })
+
   it('renders null when redirecting to the app, even while waiting for authRequestCompleted', () => {
```
Comment (kmjennison) [inline src/__tests__/withAuthUser.test.js:852]: Also change to "renders null by default"

File: src/__tests__/withAuthUser.test.js [hunk 3]
```diff
@@ -826,6 +826,29 @@ describe('withAuthUser: rendering/redirecting', () => {
...
+    })(MockComponent)
+    const { queryByText } = render(
+      <MockCompWithUser
+        serializedAuthUser={MockSerializedAuthUser}
+        message="How are you?"
+      />
+    )
+    expect(queryByText('Hello! How are you?')).toBeTruthy()
+  })
+
```
Comment (kmjennison) [inline src/__tests__/withAuthUser.test.js:851]: Please add a test for when `whenAuthedBeforeRedirect` is set to `AuthAction.SHOW_LOADER`.

**Review by kmjennison — CHANGES_REQUESTED** [review]
Thank you for the comprehensive PR! Some minor changes requested.

**Comment by abusada** [comment]
@kmjennison updated the PR according to your review

**Review by kmjennison — APPROVED** [review]
Looks great, thanks!

**Pull request merged**